### PR TITLE
Test Cleanup: Fix Assert.NotNulls with ModelBindingResult

### PR DESCRIPTION
**Pull request by rynowak**
MBR is a struct now, updated tests and names to reflect this. These were missed in an earlier pass.

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/BindingSourceModelBinderTest.cs [hunk 1]
```diff
@@ -90,7 +90,7 @@ public void BindingSourceModelBinder_ThrowsOnNonGreedySource()
             var result = await binder.BindModelAsync(context);
 
             // Assert
-            Assert.NotNull(result);
+            Assert.NotEqual(ModelBindingResult.NoResult, result);
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/BindingSourceModelBinderTest.cs]: This test is failing in the CI builds. Probably now two separate tests: In one case the system returns `NoResult` and in the other, the model is set.

**Comment by dougbu** [comment]
:shipit: after getting the CI builds working

**Comment by davidfowl** [comment]
Why is travis all red on MVC all of a sudden?

**Comment by NeelBhatt** [comment]
I guess here something is wrong: 

https://github.com/aspnet/Mvc/commit/78c9752622526c6faa71193f0250d03a76c971cbBindingSource

for this test:

ModelBinder_ReturnsNonEmptyResult_MatchingSource

**Comment by dougbu** [comment]
@NeelBhatt is correct. See my [earlier comment](https://github.com/aspnet/Mvc/pull/3090#discussion-diff-38721427).

File: test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/BindingSourceModelBinderTest.cs (hunk 1 above)
Comment (NeelBhatt) [inline test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/BindingSourceModelBinderTest.cs]: I just noticed you already wrote this comment!

**Pull request closed without merging**